src/dispatch/plugins/dispatch_slack/plugin.py

```diff
@@ -447,16 +436,10 @@ def fetch_events(
             event = self.get_event(plugin_event)
             if event is None:
                 raise ValueError(f"No event found for Slack plugin event: {plugin_event}")
-            if isinstance(event, type):
-                # It's a class, instantiate it
-                event = event(plugin_event)
-            return event.fetch_activity(
-                client=client,
-                subject=subject,
-                oldest=oldest,
-            )
-        except Exception:
-            logger.exception("Error fetching events from Slack")
+            activity = event.fetch_activity(client, subject, oldest)
```

Comment:
If `event` is a class we will get the `ThreadActivityEvent.fetch_activity() missing 1 required positional`